common/lobbies/slot.ts

```diff
@@ -2,8 +2,19 @@ import cuid from 'cuid'
 import { Record } from 'immutable'
 import { RaceChar } from '../races'
 
+export enum SlotType {
+  Human = 'human',
+  Observer = 'observer',
+  Computer = 'computer',
+  ControlledOpen = 'controlledOpen',
+  ControlledClosed = 'controlledClosed',
+  UmsComputer = 'umsComputer',
+  Open = 'open',
+  Closed = 'closed',
+}
+
 export class Slot extends Record({
-  type: 'open',
+  type: undefined as SlotType | undefined,
```

Comment:
Why make this potentially undefined? There is never a point where we expect a Slot to not have a known type, and in the previous iteration of this code that default type was 'open'. `type: SlotType.Open` would be the correct mapping of the previous code